lightning/src/ln/channelmanager.rs

```diff
@@ -2938,8 +2942,34 @@ macro_rules! handle_error {
 ///
 /// Note that this step can be skipped if the channel was never opened (through the creation of a
 /// [`ChannelMonitor`]/channel funding transaction) to begin with.
-macro_rules! update_maps_on_chan_removal {
-	($self: expr, $peer_state: expr, $channel_context: expr) => {{
+macro_rules! locked_close_channel {
+	($self: ident, $peer_state: expr, $channel_context: expr, $shutdown_res_mut: expr) => {{
+		if let Some((counterparty_node_id, funding_txo, channel_id, update)) = $shutdown_res_mut.monitor_update.take() {
+			if $self.background_events_processed_since_startup.load(Ordering::Acquire) {
+				handle_new_monitor_update!($self, funding_txo, update, $peer_state,
+					$channel_context, REMAIN_LOCKED_UPDATE_ACTIONS_PROCESSED_LATER);
+			} else {
+				// We want to track the in-flight update both in `in_flight_monitor_updates` and in
+				// `pending_background_events` to avoid a race condition during
+				// `pending_background_events` processing where we complete one
+				// `ChannelMonitorUpdate` (but there are more pending) but we conclude that all
```

Comment:
I think something like this would clarify this comment for me:
```suggestion
				// `ChannelMonitorUpdate` (but there are more pending in the background events) but we conclude that all
```